paddle/fluid/operators/mkldnn/requantize_mkldnn_op.cc

```diff
@@ -56,101 +57,56 @@ class ReQuantOpKernel : public framework::OpKernel<T> {
         platform::errors::InvalidArgument("Scale of output cannot be 0.0"));
     if (shift_in != 0.0f) {
       PADDLE_ENFORCE_EQ(
-          framework::TransToProtoVarType(input->dtype()),
-          framework::proto::VarType::UINT8,
+          input->dtype(),
+          DataType::UINT8,
           platform::errors::Unimplemented("Requantize does not support nonzero "
                                           "shift for signed input."));
     }
 
     auto& dev_ctx =
         ctx.template device_context<platform::MKLDNNDeviceContext>();
-    const auto& engine = dev_ctx.GetEngine();
 
     auto src_tz = phi::vectorize(input->dims());
 
-    float reorder_scale = scale_out / scale_in;
+    auto src_paddle_dt = input->dtype();
+    auto dst_paddle_dt = with_shift ? DataType::UINT8 : src_paddle_dt;
+
+    auto xstrides = input->mem_desc().data.format_desc.blocking.strides;
+    std::vector<dnnl_dim_t> vstrides(xstrides,
+                                     xstrides + input->mem_desc().data.ndims);
 
-    std::string key = platform::CreateKey(
```

Comment:
Is the caching for requantize removed because the caching is managed internally in dnnl now?

Reply:
Yes. oneDNN caching is much more generic than ours although PaddlePaddle caching may perform better in some situations.